modules/identities/scripts/sign_in_or_sign_up.ts

```diff
@@ -0,0 +1,38 @@
+import { RuntimeError, ScriptContext } from "../module.gen.ts";
+import { IdentityDataInput, IdentityProviderInfo } from "../utils/types.ts";
+
+export interface Request {
+    info: IdentityProviderInfo;
+    uniqueData: IdentityDataInput;
+    additionalData: IdentityDataInput;
+
+    username?: string;
+}
+
+export interface Response {
+    userToken: string;
+}
+
+export async function run(
+	ctx: ScriptContext,
+	req: Request,
+): Promise<Response> {
+    try {
```

Comment:
do this all inside a single script without script calls so we can do everything within a single tx to prevent race conditions